src/polyswarm_api/types/result.py

```diff
@@ -47,10 +47,18 @@ def __init__(self, result, polyswarm=None):
         self.order_by = json.get('order_by', None)
         self.direction = json.get('direction', None)
 
+        # This are set by subclasses during parsing of the API response
+        self.failed = False
+        self.failure_reason = ''
+
     @property
     def _bad_status_message(self):
         return "Got unexpected result code: {}, message: {}".format(self.status_code, self.result)
 
+    def _set_failure(self, reason='Unspecified error occurred'):
```

Comment:
I don't understand why we are storing failures. We should raise exceptions (with the necessary failure information, if needed) and let it bubble, specially since this is a python package intended to be used in other python projects. Like this, we force the user to make an error check after the method/function call, which should be avoided.